src/main/java/net/minecraftforge/common/capabilities/CapabilityManager.java

```diff
@@ -55,30 +54,56 @@
      * @param type The class type to be registered
      * @deprecated use {@link RegisterCapabilitiesEvent}
      */
-    @Deprecated(forRemoval = true)
+    @Deprecated(since = "1.18", forRemoval = true)
     public <T> void register(Class<T> type)
     {
         Objects.requireNonNull(type,"Attempted to register a capability with invalid type");
-        String realName = type.getName().intern();
+        get(Type.getType(type).toString(), true);
+    }
+
+
+    public static <T> Capability<T> get(CapabilityToken<T> type)
+    {
+        return INSTANCE.get(type.getType(), false);
```

Comment:
`CapabilityToken#getType` returns the internal name (using slashes), all other callers of `get(String, boolean)` pass the binary name (using dots) though.